crates/pipeline-manager/src/db/operations/pipeline.rs

```diff
@@ -1357,6 +1359,102 @@ pub(crate) async fn list_pipeline_programs_across_all_tenants(
         .collect())
 }
 
+/// Store a support data collection entry
+pub(crate) async fn store_support_data_collection(
+    txn: &Transaction<'_>,
+    pipeline_id: PipelineId,
+    tenant_id: TenantId,
+    support_bundle: &SupportBundleData,
+) -> Result<(), DBError> {
+    let query = r#"
+        INSERT INTO support_data_collections
+        (pipeline_id, tenant_id, collected_at, support_bundle_data)
+        VALUES ($1, $2, $3, $4)
+    "#;
+
+    // Serialize the support bundle data using MessagePack
+    let serialized_data = to_vec(support_bundle).map_err(|e| DBError::InvalidJsonData {
+        data: "support_bundle_data".to_string(),
+        error: format!("Failed to serialize support bundle data: {}", e),
+    })?;
+
+    let stmt = txn.prepare_cached(query).await?;
+    txn.execute(
+        &stmt,
+        &[
+            &pipeline_id.0,
+            &tenant_id.0,
+            &support_bundle.time.naive_utc(),
+            &serialized_data,
+        ],
+    )
+    .await?;
+
+    Ok(())
+}
+
+/// Clean up old support data collections for a pipeline.
+///
+/// Returns the amount of deleted collections.
+pub(crate) async fn cleanup_old_support_data_collections(
```

Comment:
may this leave support bundles for old pipelines forever in the database?

Reply:
added tests for this